astropy/visualization/interval.py

```diff
@@ -226,9 +229,10 @@ class ZScaleInterval(BaseInterval):
         5.
     """
 
-    def __init__(self, nsamples=1000, contrast=0.25, max_reject=0.5,
+    @deprecated_renamed_argument('nsamples', 'n_samples', '5.2')
+    def __init__(self, n_samples=1000, contrast=0.25, max_reject=0.5,
                  min_npixels=5, krej=2.5, max_iterations=5):
-        self.nsamples = nsamples
+        self.n_samples = n_samples
```

Comment:
Do we need something like this for backward compatibility? Though it can be confusing if user modifies `self.nsamples` and not `self.n_samples`. But without this, the code breaks.
```suggestion
        self.n_samples = n_samples
        self.nsamples = self.n_samples
```

Reply:
It is possible. Maybe also emit a deprecation warning in that setter?

Reply:
Agreed. Should the `getter` also have a deprecation warning as well?

Reply:
You should be using the `deprecated_attribute` decorator here: https://github.com/astropy/astropy/blob/f298b8bc98c44104d183d56bd67f541f7d273f47/astropy/utils/decorators.py#L206

Reply:
The old keyword/attribute are set and got via `@deprecated` `@property` methods now.

Reply:
> You should be using the `deprecated_attribute` decorator here:
> 
> https://github.com/astropy/astropy/blob/f298b8bc98c44104d183d56bd67f541f7d273f47/astropy/utils/decorators.py#L206

I tried using that decorator, but got
```
TypeError: 'property' object is not callable
```
errors... the simple `@deprecated` did not yield such errors.  Suggestions?

Reply:
Yes, but you have reinvented the wheel.

Reply:
I do that a lot.  Suggested code edits are welcome.

Reply:
I shouldn't have called `deprecated_attribute()` a decorator because it isn't. Furthermore, using it to rename an attribute is not as convenient as it should be. I have opened #13824 to make the function simpler to use. It might be a good idea to rebase this pull request once the other one is merged to take advantage of the new functionality.

Reply:
Now using ``deprecated_attribute()`` to access the new attribute using the old attribute name.